internal/pkg/aws/ec2/ec2.go

```diff
@@ -0,0 +1,125 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package ec2 provides a client to make API requests to Amazon Elastic Compute Cloud.
+package ec2
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+)
+
+var (
+	fmtFilterTagApp = fmt.Sprintf("tag:%s", deploy.AppTagKey)
+	fmtFilterTagEnv = fmt.Sprintf("tag:%s", deploy.EnvTagKey)
+	defaultFilter   = ec2.Filter{
+		Name:   aws.String(filterDefault),
+		Values: aws.StringSlice([]string{"true"}),
+	}
+)
+
+const (
+	filterDefault = "default-for-az"
+)
+
+type api interface {
+	DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error)
+	DescribeSecurityGroups(*ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error)
+}
+
+type EC2 struct {
```

Comment:
Can we add doc comments to all exported names in the package? https://github.com/golang/go/wiki/CodeReviewComments#doc-comments